packages/admin-api/src/controllers/notification.controller.ts

```diff
@@ -0,0 +1,94 @@
+        }
+        const { email } = req.kauth.grant.access_token.content
+        if (type !== "application" && type !== "claim") {
+            return res.status(400).send("Invalid type")
+        }
+        const username = `${req.kauth.grant.access_token.content.idp_username}@${req.kauth.grant.access_token.content.identity_provider}`
+        const notifications = await notificationService.getNotification(email, Number(catchmentNo), type, username)
+        return res.status(200).send(notifications)
+    } catch (e: unknown) {
+        console.error(e)
+        return res.status(500).send("Internal Server Error")
+    }
+}
+
+export const addNotification = async (req: any, res: express.Response) => {
+    try {
+        const { catchmentNo, type } = req.body
+        const catchments = await getCatchments(req.kauth.grant.access_token)
+
+        if (catchments.length === 0 || !catchments.includes(Number(catchmentNo))) {
+            return res.status(403).send("Forbidden")
+        }
+
+        const { email } = req.kauth.grant.access_token.content
+        if (type !== "application" && type !== "claim") {
+            return res.status(400).send("Invalid type")
+        }
+        const username = `${req.kauth.grant.access_token.content.idp_username}@${req.kauth.grant.access_token.content.identity_provider}`
+        const result = await notificationService.addNotification(email, Number(catchmentNo), type, username)
+        return res.status(200).send(result)
+    } catch (e: unknown) {
+        console.error(e)
+        return res.status(500).send("Internal Server Error")
+    }
+}
+
+export const deleteNotification = async (req: any, res: express.Response) => {
+    try {
+        const { catchmentNo, type } = req.body
+        const catchments = await getCatchments(req.kauth.grant.access_token)
+
+        if (catchments.length === 0 || !catchments.includes(Number(catchmentNo))) {
+            return res.status(403).send("Forbidden")
+        }
+
+        const { email } = req.kauth.grant.access_token.content
+        if (type !== "application" && type !== "claim") {
+            return res.status(400).send("Invalid type")
+        }
+        const username = `${req.kauth.grant.access_token.content.idp_username}@${req.kauth.grant.access_token.content.identity_provider}`
+        const result = await notificationService.deleteNotification(email, Number(catchmentNo), type, username)
+        return res.status(200).send(result)
+    } catch (e: unknown) {
+        console.error(e)
+        return res.status(500).send("Internal Server Error")
+    }
+}
+
+export const checkNotificationEmail = async (req: any, res: express.Response) => {
+    try {
+        const username = `${req.kauth.grant.access_token.content.idp_username}@${req.kauth.grant.access_token.content.identity_provider}`
+        const notifications = await notificationService.getAllNotificationsFromUser(username)
+        await Promise.all(
```

Comment:
I think this update should be performed in a single operation, maybe an updateNotifications() function in the notifications service. It could trigger a large number of requests and will perform a partial update if any of the requests fail.

Reply:
Actually, it looks like knex doesn't support batch updates, so multiple requests are required:
https://github.com/knex/knex/issues/5716
May be preferable to perform these requests in a transaction to make them 'all or none', as was done for catchment changes in the admin API. However, that behaviour isn't as critical here as is it for catchment changes. Resolving this comment, and may add backlog item to review our batch updates and implement transactions as necessary.